python/ray/data/dataset.py

```diff
@@ -5013,7 +5010,7 @@ def names(self) -> List[str]:
         return self.base_schema.names
 
     @property
-    def types(self) -> List[Union[Literal[object], "pyarrow.DataType"]]:
+    def types(self) -> List[Union[object, "pyarrow.lib.DataType"]]:
```

Comment:
Got it, thanks for catching this. I've changed `Type[object] -> type[object]` to avoid needing to import it; `Type` is only needed for python<3.8, which we no longer support.